frontend_app_share_folder_advanced/src/component/NewUpload.jsx

```diff
@@ -79,39 +84,49 @@ class NewUpload extends React.Component {
-              placeholder={props.t('Password to share link (optional)')}
-              value={props.uploadPassword}
-              onChange={props.onChangeUploadPassword}
-            />
-            <button
-              type='button'
-              className='newUpload__password__icon'
-              key='seeuploadPassword'
-              title={props.t('Show password')}
-              style={{ ':hover': { color: customColor } }}
-              data-cy='seePassword'
-              onClick={this.handleTogglePasswordVisibility}
-            >
-              <i className={state.hidePassword ? 'fa fa-fw fa-eye' : 'fa fa-fw fa-eye-slash'} />
-            </button>
-          </div>
-          <button
-            type='button'
-            className='newUpload__password__icon'
-            key='randomuploadPassword'
-            title={props.t('Generate random password')}
-            style={{ ':hover': { color: customColor } }}
-            onClick={this.handleRandomPassword}
-          >
-            <i className='fa fa-fw fa-repeat' />
-          </button>
-        </div>
+        {state.isPasswordActive
+          ? (
+            <div className='newUpload__password'>
+              <div className='newUpload__password__wrapper'>
+                <i className='fa fa-fw fa-lock' />
+                <input
+                  type={state.hidePassword ? passwordType : textType}
+                  className='newUpload__password__input form-control'
+                  placeholder={props.t('Password to share link (optional)')}
+                  value={props.uploadPassword}
+                  onChange={props.onChangeUploadPassword}
+                />
+                <button
+                  type='button'
+                  className='newUpload__password__icon'
+                  key='seeuploadPassword'
+                  title={props.t('Show password')}
+                  style={{ ':hover': { color: customColor } }}
+                  data-cy='seePassword'
+                  onClick={this.handleTogglePasswordVisibility}
+                >
+                  <i className={state.hidePassword ? 'fa fa-fw fa-eye' : 'fa fa-fw fa-eye-slash'} />
+                </button>
+              </div>
+              <button
+                type='button'
+                className='newUpload__password__icon'
+                key='randomuploadPassword'
+                title={props.t('Generate random password')}
+                style={{ ':hover': { color: customColor } }}
+                onClick={this.handleRandomPassword}
+              >
+                <i className='fa fa-fw fa-repeat' />
+              </button>
+            </div>
+          )
+          : (
+            <div className='newUpload__password'>
+              <span className='newUpload__password__link' onClick={this.handleClickSeePasswordInput}>
+                {props.t('Protect by password')}
+              </span>
+            </div>
+          )}
```

Comment:
the closing bracket should be at the next line since the opening ones are on 2 different lines